firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/Datagrams.cpp

```diff
@@ -810,25 +809,26 @@ SCENARIO(
       THEN(
           "The paylaod buffer returned by the paylaod accessor method is equal to the payload from "
           "the body") {
-        auto expected_payload = std::string("\xca\x9b\xc7\xf9\x5c\x10\xa1\x77\x23\x82", 10);
+        auto expected_payload = std::string("\x61\x6a\x1a\x6a\x29\xcf\x01\x81\xbe\x9d", 10);
         REQUIRE(parsed_datagram.payload() == expected_payload);
       }
       THEN(
           "the payload given in the ParsedDatagram constructor is independent of the input buffer "
           "body") {
-        input_buffer.push_back(0x02);
-        auto expected_payload = std::string("\xca\x9b\xc7\xf9\x5c\x10\xa1\x77\x23\x82", 10);
+        output_buffer.push_back(0x02);
+        auto expected_payload = std::string("\x61\x6a\x1a\x6a\x29\xcf\x01\x81\xbe\x9d", 10);
         REQUIRE(parsed_datagram.payload() == expected_payload);
       }
 
       // Write
-      auto final_data = std::string("\x56\xd6\x42\xc5\xe1\xf0\x30\xe5\xc8\x4d", 10);
+      // payload_output = parsed_datagram.payload()
+      auto payload_output = std::string("\x61\x6a\x1a\x6a\x29\xcf\x01\x81\xbe\x9d", 10);
       TestDatagramProps::PayloadBuffer input_payload;
-      PF::Util::convert_string_to_byte_vector(final_data, input_payload);
+      PF::Util::convert_string_to_byte_vector(payload_output, input_payload);
       TestConstructedDatagram write_datagram(input_payload, 1);
```

Comment:
Was there a particular reason you're doing all this array copying and making a redundant hard-coded version of the data in line 738 (so that if we change line 738 but forget to change line 825, then the test continues to pass but it no longer behaves like a round-trip test), instead of just doing ` TestConstructedDatagram write_datagram(parsed_payload, 1);` or ` TestConstructedDatagram write_datagram(parsed_datagram.payload(), 1);`?

Along these lines of maintainability, but a question of saving you work rather than about the risk of a test no longer doing what we say it does: is there a reasonable way you can define `"\x61\x6a\x1a\x6a\x29\xcf\x01\x81\xbe\x9d"` as a constant once, instead of hard-coded copy-pasting those bytes for all the REQUIREs?